blog/topology-rules.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Add non skydive nodes in topology
+layout: blog-post
+author: Masco Kaliyamoorthy
+date: 27/11/2018
+---
+
+Skydive displays the network topology by receiving the network events from the skydive agents. You ever wondered how to add or display in topology diagram, a network components which is out of the skydives’ agent network or a non network entities like TOR, data store and etc. No more worries on that, thanks to the skydive ‘Topology Rules’ API.
+
+Since version 0.20, Skydive provides Topology Rules API, can be used to create new nodes and edges and update existing nodes’ metadata. Topology Rules API divided in two APIs, node rule API and edge rule API. Node rule API is used for create a new node and update metadata of existing node. Edge rule API is used for create edge between two nodes i.e linking two nodes.
+
+In this blog we will see two use cases, one is a network component  which is not part of skydive network. Secondly, a non network component.
+
+Before that, we will see some basic usage of Topology rules API.
+
+## Create Node
+
+To create node, You must provide a unique node name and a valid node type. You can provide some optional params as well.
+
+{% highlight shell %}
+skydive client node-rule create --action="create" --node-name="node1" --node-type="fabric" --name="node rule1"
+{
+  "UUID": "ea21c30f-cfaa-4f2d-693d-95159acb71ed",
+  "Name": "node rule1",
+  "Description": "",
+  "Metadata": {
+    "Name": "node1",
+    "Type": "fabric"
+  },
+  "Action": "create",
+  "Query": ""
+}
+{% end highlight %}
+
+## Update Nodes’ Metadata
+
+To update the metadata of an existing node, you must provide a ‘gremlin query’ to chose the nodes you want to update the metadata. According to your query you can update the metadata of one or more nodes with single node rule.
+
+{% highlight shell %}
+
+skydive client node-rule create --action="update" --name="update rule" --query="G.V().Has('Name', 'node1')" --metadata="key1=val1, key2=val2"
+{
+  "UUID": "3e6c0e15-a863-4583-6345-715053ac47ce",
+  "Name": "update rule",
+  "Description": "",
+  "Metadata": {
+    "key1": "val1",
+    "key2": "val2"
+  },
+  "Action": "update",
+  "Query": "G.V().Has('Name', 'node1')"
+}
+{% end highlight %}
+
+## Create Edge
+
+To create edge, you should provide the source and destination nodes AND type of edge, the value should be either `layer2` or `ownership`.
```

Comment:
In fact the relationtype can be anything except that for ownership we do something specific in the webui. I think it is worth if to explain that.

Reply:
rephrased the relation type with the new change.